website/docs/cloud-docs/architectural-details/security-model.mdx

```diff
@@ -116,6 +116,16 @@ Marking a variable as “sensitive” will prevent it from being displayed in th
 
 The logs from a Terraform plan or apply operation are visible to any user with at least “read” level access in the associated workspace. While Terraform tries to avoid writing sensitive information to logs, redactions are best-effort. This feature should not be treated as a security boundary, but instead as a mechanism to mitigate accidental exposure. Additionally, HCP Terraform is unable to protect against malicious users who attempt to use Terraform logs to exfiltrate sensitive data.
 
+### Redaction of ephemeral values in Terraform logs
+
+The logs from a Terraform plan or apply operation are visible to any user with at least “read” level access in the associated workspace. Terraform tries to avoid writing ephemeral values to logs
+and provider developers are discouraged from logging those as well but Terraform cannot provide guarantees that providers will not log ephemeral values.
+You can reduce the risk by only [using trusted modules and providers within Terraform configuration](#malicious-terraform-providers-or-modules).
+
+### Redaction of ephemeral values in memory
+
+Ephemeral Values prevents values from being persisted to disk (as part of a plan file or state file) but no efforts are made to protect ephemeral values from memory analysis of the running application.
```

Comment:
Is there any recommendation we can give to folks here? Is this something they need to worry about? Or more like something we are giving them a heads-up about?
```suggestion
Terraform does not persist ephemeral values to plan or state files. However, Terraform does not protect ephemeral values from a memory analysis of your running application.
```